elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/ElasticsearchIndexManager.java

```diff
@@ -458,28 +462,34 @@ private JsonObject addFieldMapping(JsonObject payload, FacetMetadata facetMetada
 	}
 
 	private String getIndex(EntityIndexBinding binding, DocumentFieldMetadata fieldMetadata) {
-		// Never analyze boolean, date or numeric
+		String index = getIndexValue( fieldMetadata.getIndex() );
+
+		// Never analyze booleans, numerics, dates
 		if ( FieldHelper.isBoolean( binding, fieldMetadata.getName() ) ||
 				FieldHelper.isDate( binding, fieldMetadata.getName() ) ||
 				FieldHelper.isCalendar( binding, fieldMetadata.getName() ) ||
 				FieldHelper.isNumeric(fieldMetadata) ) {
-			return NOT_ANALYZED;
+			if ( ANALYZED.equals( index ) ) {
+				return NOT_ANALYZED;
+			}
 		}
 
-		return getIndexValue( fieldMetadata.getIndex() );
+		return index;
 	}
 
-	@SuppressWarnings("deprecation")
 	private String getIndex(BridgeDefinedField bridgeDefinedField) {
-		// Never analyze boolean
+		String index = getIndexValue( bridgeDefinedField.getIndex() );
+
+		// Never analyze booleans, numerics, dates
 		if ( FieldHelper.isBoolean( bridgeDefinedField )
 				|| FieldHelper.isNumeric( bridgeDefinedField )
 				|| FieldHelper.isDate( bridgeDefinedField ) ) {
-			return NOT_ANALYZED;
+			if ( ANALYZED.equals( index ) ) {
+				return NOT_ANALYZED;
+			}
 		}
 
-		Field.Index index = bridgeDefinedField.getIndex();
-		return getIndexValue( index );
+		return index;
 	}
 
 	private String getIndexValue(Field.Index index) {
```

Comment:
Note: this comment is mostly orthogonal to your patch but I think it could benefit from it.

You will probably do that when you'll fix the norms issue but I'm wondering if we should introduce a private `IndexOptions` class with all these information.

I think I would prefer if we pass an additional `canBeAnalyzed` boolean to `getIndexValue` and keep all the logic in getIndexValue instead of changing the value in the respective `getIndex`. This would allow to return an immutable `IndexOptions` from `getIndexValue` (which would be renamed `getIndexOptions`).

In `IndexOptions`, we would have `getIndexValue()`, `isIndexed()` and in the future `hasNorms()`.